cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -662,935 +695,1030 @@ fn valid_for_target(triple: &Triple, op: Opcode, args: &[Type], rets: &[Type]) -
+
+static OPCODE_SIGNATURES: Lazy<Vec<OpcodeSignature>> = Lazy::new(|| {
+    let types = &[
+        I8, I16, I32, I64, I128, // Scalar Integers
+        F32, F64, // Scalar Floats
+        I8X16, I16X8, I32X4, I64X2, // SIMD Integers
+        F32X4, F64X2, // SIMD Floats
+    ];
+
+    Opcode::all()
+        .iter()
+        .filter(|op| {
+            match op {
+                // Control flow opcodes should not be generated through `generate_instructions`.
+                Opcode::BrTable | Opcode::Brif | Opcode::Jump | Opcode::Return => return false,
+
+                // These are opcodes that we haven't vetted for fuzzing.
+                Opcode::Debugtrap
+                | Opcode::Trap
+                | Opcode::Trapz
+                | Opcode::ResumableTrap
+                | Opcode::Trapnz
+                | Opcode::ResumableTrapnz
+                | Opcode::CallIndirect
+                | Opcode::ReturnCall
+                | Opcode::ReturnCallIndirect
+                | Opcode::FuncAddr
+                | Opcode::X86Pshufb
+                | Opcode::AvgRound
+                | Opcode::Uload8x8
+                | Opcode::Sload8x8
+                | Opcode::Uload16x4
+                | Opcode::Sload16x4
+                | Opcode::Uload32x2
+                | Opcode::Sload32x2
+                | Opcode::StackAddr
+                | Opcode::DynamicStackLoad
+                | Opcode::DynamicStackStore
+                | Opcode::DynamicStackAddr
+                | Opcode::GlobalValue
+                | Opcode::SymbolValue
+                | Opcode::TlsValue
+                | Opcode::GetPinnedReg
+                | Opcode::SetPinnedReg
+                | Opcode::GetFramePointer
+                | Opcode::GetStackPointer
+                | Opcode::GetReturnAddress
+                | Opcode::TableAddr
+                | Opcode::Null
+                | Opcode::X86Blendv
+                | Opcode::VallTrue
+                | Opcode::IcmpImm
+                | Opcode::X86Pmulhrsw
+                | Opcode::IaddImm
+                | Opcode::ImulImm
+                | Opcode::UdivImm
+                | Opcode::SdivImm
+                | Opcode::UremImm
+                | Opcode::SremImm
+                | Opcode::IrsubImm
+                | Opcode::IaddCin
+                | Opcode::IaddCarry
+                | Opcode::UaddOverflowTrap
+                | Opcode::IsubBin
+                | Opcode::IsubBout
+                | Opcode::IsubBorrow
+                | Opcode::BandImm
+                | Opcode::BorImm
+                | Opcode::BxorImm
+                | Opcode::RotlImm
+                | Opcode::RotrImm
+                | Opcode::IshlImm
+                | Opcode::UshrImm
+                | Opcode::SshrImm
+                | Opcode::IsNull
+                | Opcode::IsInvalid
+                | Opcode::ScalarToVector
+                | Opcode::X86Pmaddubsw
+                | Opcode::X86Cvtt2dq => return false,
```

Comment:
If we use `exceptions!` below, it probably would be fairly easy to move this block into that list, that way Its one less place to look at. (The special opcodes should probably still be rejected here)

Reply:
I'll move the ones that haven't been vetted into there, and leave the others with comments about why they're there :+1: